### PR TITLE
refactor: change codebase into material-ui

### PR DESCRIPTION
**Pull request by viinxent**
Hi everyone!

I did not change any logic on the app. The app works as it is, this branch purpose was to make sure to remove the react-bootstrap dependencies/components and move to material ui one. Also I didn't remove the packages yet for `react-bootstrap`.

The design is kind of **meh** as of now, since my main focus was to make it work for material ui.

Problem faced:

There's this warning using the stripe elements with material ui, can't seem to rid of it.
Material UI have this "hackish" way of using stripe elements.

Problem can be seen at this commit: https://github.com/MangaSeed/serverless-demo-client-redux-materialui/commit/4d4bebfa16f818e5afb489996795695920abb245

See this post/issue from material ui:
https://github.com/mui-org/material-ui/issues/16037

### CODE REVIEW

File: src/components/BillingForm.js [hunk 1]
```diff
@@ -1,76 +1,112 @@
...
-import "./BillingForm.css";
+import React, { useState } from 'react';
+import { CardElement, injectStripe } from 'react-stripe-elements';
+import { TextField, Divider } from '@material-ui/core';
+
+import LoaderButton from './LoaderButton';
+
+import { useFormFields } from '../libs/hooksLib';
+
+function StripeInput(props) {
```
Comment (xuatz) [inline src/components/BillingForm.js:9]: What's happening here?

File: src/containers/Home/Home.js [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+import { useHomeStyle } from './Home.style';
+
+export default function Home(props) {
+  const classes = useHomeStyle();
+  const [notes, setNotes] = useState([]);
+  const [isLoading, setIsLoading] = useState(true);
+
+  useEffect(() => {
+    async function onLoad() {
```
Comment (xuatz) [inline src/containers/Home/Home.js:22]: why do you need to create new `onLoad()` every time?

File: src/containers/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+      setIsLoading(false);
+    }
+
+    onLoad();
+  }, [props.isAuthenticated]);
+
+  function loadNotes() {
+    return API.get('notes', '/notes');
+  }
```
Comment (xuatz) [inline src/containers/Home/Home.js:42]: could move this out of the component

**Review by xuatz — APPROVED** [review]
Let's apply the changes after the typescript migration.

File: src/containers/Home/Home.js (hunk 2 above)
Comment (viinxent) [reply src/containers/Home/Home.js:22]: Hey @xuatz

Sorry, I didn't change the logic of the application boilerplate currently. So there are questionable logic in the current state of the app. They will be moved out after I integrated `redux/redux-saga`

File: src/components/BillingForm.js (hunk 1 above)
Comment (viinxent) [reply src/components/BillingForm.js:9]: Hey @xuatz

I was trying to integrate the `react-stripe-elements` into `material-ui` unfortunately they don't have a clean way of doing it. One of the author of `material-ui` created a `snippet`. `TextField` component of `material ui` have a `InputProps` property which in a way you could override the default component by passing a custom props. It's not actually documented on their web unfortunately.

But they have a topic for it here:
https://github.com/mui-org/material-ui/issues/16037

**Pull request merged**